src/Symfony/Component/Ldap/LdapClient.php

```diff
@@ -86,6 +86,8 @@ public function find($dn, $query, $filter = '*')
         $search = ldap_search($this->connection, $dn, $query, $filter);
         $infos = ldap_get_entries($this->connection, $search);
 
+        ldap_free_result($search);
```

Comment:
`ldap_free_results` won't work if the `ldap_search` method call returns `false` or throws an error.

You should add some error handling code around the statement.